pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -1090,7 +1124,23 @@ def clear(self):
         self.scatter.clear()
 
     def appendData(self, *args, **kargs):
-        pass
+        """
+        Append to displayed data.
+        See :func:`__init__() <pyqtgraph.PlotDataItem.__init__>` for details; it accepts the same arguments.
+        In addition to the listed data formats, appendData also accepts single numerical values as 
+        ==========================  ======================================
+        appendData(xValue, yValue)  x and y values may be int or float
+        appendData(yValue)          y value may be int or float
+        ==========================  ======================================
+        """
+        arg_list = list(args)
+        for idx in range(len(args)):
+            if idx == 2: break # only convert the first two arguments
+            try: # if it converts to float, then convert to single-value ndarray
+                arg_list[idx] = np.full(1, float(arg_list[idx]))
+            except TypeError: pass # TypeError occurs for data that does not need conversion
```

Comment:
We can clean this up a decent amount I think!

```python
# at the top of the file
from contextlib import suppress


....

def appendData(self, *args, **kwargs):

    arg_list = list(args)
    for idx, arg in enumerate(args):
        if idx == 2:
            break
        with suppress(TypeError):
            # TypeError occurs for data that does not need conversion
            arg_list[idx] = np.full(1, float(arg))
         kargs['append'] = True
         self.setData(*arg_list, **kargs)
```

`for _ in range(len(iterable))` is generally considered a code-smell, if you care about indexes you should use `enumerate(iterable)`.  Also whenever you have try/except: pass, the [`suppress`](https://docs.python.org/3/library/contextlib.html#contextlib.suppress) is typically what you want; functionally they're the same thing, but it's just a bit more readable ...but now we're in nit-pick territory.